.gitattributes

```diff
@@ -52,20 +52,22 @@ README*                  ocaml-typo=missing-header
 /tools/ci/appveyor/appveyor_build.cmd      ocaml-typo=long-line,very-long-line,missing-header,non-ascii text eol=crlf
 /tools/ci/appveyor/appveyor_build.sh       ocaml-typo=non-ascii
 
+# depend files are auto-generated
+.depend                  ocaml-typo=missing-header
 
 asmcomp/*/emit.mlp       ocaml-typo=tab,long-line,unused-prop
 asmcomp/power/NOTES.md   ocaml-typo=missing-header
 
-/config/gnu              ocaml-typo=prune
+/config/gnu**            ocaml-typo=prune
```

Comment:
This (and the other two changes) are slightly in conflict with the reasoning behind https://github.com/ocaml/ocaml/pull/1148/commits/2adc6af92a60815f76260928ae2829336bf2ade3 - not necessarily a problem. The alternative is to steal the logic from https://github.com/ocaml/ocaml/blob/trunk/tools/ci/travis/travis-ci.sh#L187-L203

Reply:
I don't like the logic from travis-ci.sh; I don't think that the consumers of `check-typo` should each reproduce logic that is easily expressed in it or in gitattributes. (We could have the recurse-on-dirname logic in check-typo itself, but it would probably not help with the slowness of it).

The reasoning is "have the attributes apply to exactly the right files, no more". I would say that taking the transitive closure of a parent directory is not really a loss of precision, but a different choice of semantics. On the other hand, it may be that with the current formulation `/config/gnufoo` would also be concerned; I can test and, if this is the case, change it to have both `/config/gnu` and `/config/gnu/**`.

On the other hand, one option is to say "I want to be able to run check-typo on manual files, when running the tool from the manual directory itself for example, just they shouldn't be traversed on runs from the root". Then the change would be clearly wrong. But I don't think that we want that -- either we decide to keep a file under strict discipline, or we should give up checking on it all the time, I don't see a meaningful compromise in-between.